internal/docstore/docstore.go

```diff
@@ -260,10 +260,14 @@ func (e ActionListError) Unwrap() error {
 	return nil
 }
 
-// BeforeDo takes a callback function that will be called before the ActionList
-// is executed by the underlying provider's action functionality. The callback
-// takes a parameter, asFunc, that converts its argument to provider-specific
-// types. See https://gocloud.dev/concepts/as/ for background information.
+// BeforeDo takes a callback function that will be called before the ActionList is
+// executed by the underlying provider. It may be invoked multiple times for a
+// single call to ActionList.Do. If the callback returns an error, ActionList.Do
```

Comment:
When would it be invoked multiple times? Consider explaining this.

Reply:
Done.